lib/modules/datasource/go/index.ts

```diff
@@ -24,6 +24,10 @@ export class GoDatasource extends Datasource {
   readonly goproxy = new GoProxyDatasource();
   readonly direct = new GoDirectDatasource();
 
+  // Pseudo versions https://go.dev/ref/mod#pseudo-versions
+  static readonly pversionRegexp =
+    /v\d+\.\d+\.\d+-(?:(?:\w+\.)?0\.)?\d{14}-([a-f0-9]{12})/;
```

Comment:
```suggestion
    regEx(/v\d+\.\d+\.\d+-(?:\w+\.)?(?:0\.)?\d{14}-(?<digest>[a-f0-9]{12})/);
```

please use our regex wrapper and use named capture group
I also reduced nesting, should probably a little bit better in perf